### PR TITLE
[RETINA-210] NFC Support

### LINKED ISSUE #210: NFC support

### PR DESCRIPTION
**Pull request by Jbcampbe**
Closes #210 #244

### CODE REVIEW

File: src/components/select.vue [hunk 1]
```diff
@@ -1140,19 +1144,9 @@ export default {
...
-    background: #5897fb;
-    color: #fff;
-  }
-  .v-select .highlight:not(:last-child) {
-    margin-bottom: 0; /* Fixes Bulma Margin */
-  }
+
   /* Loading Spinner */
-  .v-select .spinner {
+  /* .v-select .spinner {
```
Comment (ekraihan) [inline src/components/select.vue]: More commented out code here and below....

File: src/routes/new-tool.vue [hunk 2]
```diff
@@ -164,6 +165,7 @@
         <v-date-picker
           v-model="purchaseDate"
```
Comment (ekraihan) [inline src/routes/new-tool.vue]: Looks like there is currently no validation for purchaseDate?

**Comment by ekraihan** [comment]
Jo, I wondered if you could not show the "Scan" button on the tools page (and maybe add tool page?) when you are on a computer browser (if there is any way to detect that)?

**Comment by ekraihan** [comment]
Also, just a note on the "scan" buttons... They look kind of "disabled" to me because they are light gray; "grayed out" is how it strikes me. Maybe they could be darker?

File: src/components/nfc-encode.vue [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+        this.$toasted.show('An Error Occurred Trying to Write Tag. Please Try Again or Contact Support')
+      }
+
+      this.pauseNfcListener()
+      this.$modal.hide('ready-to-scan-modal')
+    },
+
+    _nfcCallback (tag) {
+      const record = [
+        window.ndef.textRecord(`${this.toolId} - Property of Renascent inc. (http://renascentinc.com)`)
```
Comment (ekraihan) [inline src/components/nfc-encode.vue]: Renascent, Inc.  (with a comma and capital "I" ?) From the website:

image

Comment (ekraihan) [reply src/components/nfc-encode.vue]: Also I didn't know if you could prepend the string with "Tool"? So

`Tool #${this.toolId} - Property of Renascent, Inc. (http://renascentinc.com)` ?

Just an idea, I know that would make parsing out the number harder :)

Comment (ekraihan) [reply src/components/nfc-encode.vue]: Also, we'll want to make this dynamic if we ever go public...

Comment (Jbcampbe) [reply src/components/nfc-encode.vue]: I didn't feel like we needed to clarify the tool id at all. Someone should only be seeing that in a lost or stolen type of situation in which case they don't care what the tool id is.

File: src/routes/new-tool.vue (hunk 2 above)
Comment (Jbcampbe) [reply src/routes/new-tool.vue]: will be fixed with the #235 work

File: src/assets/icons/svg/nfc.js [hunk 4]
```diff
@@ -0,0 +1,10 @@
+/* eslint-disable */
+var icon = require('vue-svgicon')
+icon.register({
+  'nfc': {
+    width: 24,
+    height: 24,
+    viewBox: '0 0 24 24',
+    data: '<path pid="0" d="M4 20h16V4H4v16z" _fill="none"/><path pid="1" d="M20 2H4c-1.1 0-2 .9-2 2v16c0 1.1.9 2 2 2h16c1.1 0 2-.9 2-2V4c0-1.1-.9-2-2-2zm0 18H4V4h16v16zM18 6h-5c-1.1 0-2 .9-2 2v2.28c-.6.35-1 .98-1 1.72 0 1.1.9 2 2 2s2-.9 2-2c0-.74-.4-1.38-1-1.72V8h3v8H8V8h2V6H6v12h12V6z"/><path pid="2" d="M0 0h24v24H0z" _fill="none"/>'
```
[the remark this answers is not in the document]
Comment (Jbcampbe) [reply src/assets/icons/svg/nfc.js]: autogenerated : )

File: src/App.vue [hunk 5]
```diff
@@ -3,6 +3,46 @@
     <transition>
       <router-view></router-view>
     </transition>
+
+    <modal
```
[the remark this answers is not in the document]
Comment (Jbcampbe) [reply src/App.vue]: I could.... the modals don't have any logic of their own so it didn't bother me having them in here... I'll create a tech debt ticket for it

**Comment by Jbcampbe** [comment]
> Also, just a note on the "scan" buttons... They look kind of "disabled" to me because they are light gray; "grayed out" is how it strikes me. Maybe they could be darker?

That's the idea. If you don't have NFC capabilities they show up as disabled looking. And if you tap on them you get an explanation of why

**Comment by Jbcampbe** [comment]
> Jo, I wondered if you could not show the "Scan" button on the tools page (and maybe add tool page?) when you are on a computer browser (if there is any way to detect that)?

I was a little back and forth on this. One thing I wanted to make sure we did in this version of the app is to be really clear about WHY you can't do something. Instead of something just being poof gone we show a disabled action and explain why they can't perform that action.... so I wasn't sure if removing the button on web would be confusing at all

**Pull request merged**